### ISSUE TITLE
bug: 4.6.x - Pipette offset calibration for new pipettes are not saved after successfully completing the calibration process

### ISSUE THREAD

**Issue by Matt-Zwimpfer** [Opentrons/opentrons#8576]
# Overview
Switching in a new pipette onto an OT-2 update to software version 4.6.0 or greater and then successfully completing pipette offset calibration for the pipette doesn't save the calibration for the pipette. Instead, the App says the pipette is still missing pipette offset calibration.

# Steps to reproduce

1. Update OT-2 and App to 4.6.0 or greater
2. Install a new pipette on the OT-2 
3. Enter into pipette offset calibration and complete all the steps
4. Save pipette offset calibration
5. Return to Pipette tab
6. App still says that pipette offset calibration is missing for that pipette

# Current behavior
After successfully completing the pipette offset calibration steps for a new pipette, the App shows this:
![image](https://user-images.githubusercontent.com/69972997/138308517-84f26c5f-66b8-4fed-a65b-6bd3d461e321.png)

# Expected behavior
The pipette offset calibration should be saved and recognized by the App/OT-2

**Comment by Matt-Zwimpfer** [comment]
### Current Workaround
A full [factory reset](https://support.opentrons.com/en/articles/4474011-factory-resetting-the-ot-2) of calibrations (deck, pipette offset, tip length, and labware calibration) seems to resolve this error. Have also seen that a manual update is required followed by a factory reset.

**Comment by silvtal** [comment]
{"ed323db6ca1ddf197aeb20667c1a7a91c89cfb2f931f45079d483928da056812": {"tipLength": 50.90000000000004, "lastModified": "2021-11-03T08:34:34.509472+00:00", "source": "user", "status": {"markedBad": false}, "uri": "opentrons/opentrons_96_tiprack_300ul/1"}}#


Hi! Thanks for your work. I found a workaround I wanted to share.

Even after factory resetting the robot and manually upgrading this happened again - I installed a new pipette, the robot came up with the "no pipettes detected" error and after power cycling I was stuck at a calibration -> forgetting loop again.

I entered the bot's data folder via ssh and the tip length file was updated, but a "None.json" file also was created at the /data/robot/tip_lengths folder. Upon deleting it, the calibration process changed. It was as if it was really asking me to perfom tip length calibration while skipping the pipette offset calibration, because after deleting None.json the calibration process was with the tiprack only and not the calibration block used for tip length.

After that, I looked at /data/robot/tip_lengths and found out that the index.json file was like this:

{"60e107ba7e6e5d23beefe9f0c888f860f7b923c71906aaa5e7f460ab25c4f38a": ["P20SV222021031633"], "ed323db6ca1ddf197aeb20667c1a7a91c89cfb2f931f45079d483928da056812": ["P3HSV212021032905", null, "P3HMV212021040107"]}

"P3HMV212021040107" is the new pipette I was trying to set up. "P3HSV212021032905" is the one set up at the other slot. I deleted the ", null" part and recalibrated, but the robot still forgot everything.

Finally, I checked /data/robot/pipettes/left (my new pipette was at the left slot) and noticed there also was a "None.json" file there. I run "rm None.json" and the calibration data appeared **instantly** on the app screen.

**TLDR**: manually deleting /data/robot/pipettes/*/None.json seems to resolve the error.
